Assets/Python/RFCUtils.py

```diff
@@ -1146,26 +1138,24 @@ def moveSlaveToNewWorld(self, iPlayer, unit):
 		lEurope = [rBritain, rIberia, rEurope, rItaly, rScandinavia, rRussia, rBalkans, rAnatolia, rMaghreb]
 		
 		lColonies = []
-		for city in PyPlayer(iPlayer).getCityList():
-			if city.GetCy().getRegionID() not in lEurope:
+		for city in self.getCityList(iPlayer):
+			if city.getRegionID() not in lEurope:
 				lColonies.append(city)
 				
 		if len(lColonies) == 0: return
 		
-		iRand = gc.getGame().getSorenRandNum(len(lColonies), 'random colony')
-		city = lColonies[iRand].GetCy()
+		city = self.getRandomEnty(lColonies)
 		
 		unit.setXYOld(city.getX(), city.getY())
 		
 	def clearSlaves(self, iPlayer):
-		for x in range(124):
-			for y in range(68):
-				plot = gc.getMap().plot(x, y)
-				if plot.getOwner() == iPlayer:
-					if plot.getImprovementType() == gc.getInfoTypeForString("IMPROVEMENT_SLAVE_PLANTATION"):
-						plot.setImprovementType(gc.getInfoTypeForString("IMPROVEMENT_PLANTATION"))
-					if plot.isCity():
-						self.removeSlaves(plot.getPlotCity())
+		for (x, y) in self.getWorldPlotsList():
+			plot = gc.getMap().plot(x, y)
+			if plot.getOwner() == iPlayer:
+				if plot.getImprovementType() == iPlantation:

```

Comment:
You mixed the plantations here, slave plantations are supposed to be converted to regular plantations.